alf/bin/verify_checkpoint.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Utility to check whether checkpointed algorithm can be restored correctly.
+
+It works as the following:
+
+1. Save the config.
+2. Train the algorithm for a few iterations.
+3. Test the algorithm for a few steps and store the output of the algorithm and
+   the environment time steps.
+4. Save checkpoint.
+5. Create the algorithm using the saved config.
+6. Load checkpoint.
+7. Run the algorithm using the stored time steps.
+8. Compare the output from step 7 with the output from step 3. They should be
+   exactly same.
+
+
+The simplest way to use it is to invoke it in the following way:
+
+.. code-block:: bash
+
+    python -m alf.bin.verify_checkpoint --conf [CONF_FILE_NAME]
+
+You may want to set a different value of ``--num_train_iterations`` if your training
+does not start from beginning because of TrainerConfig.initial_collect_steps.
+You may also want to set a different value of ``--num_test_steps`` to test more steps.
+"""
+
+from absl import app
+from absl import flags
+from absl import logging
+import gin
+import os
+import tempfile
+import torch
+
+import alf
+from alf.trainers import policy_trainer
+from alf.algorithms.data_transformer import create_data_transformer
+from alf.environments.utils import create_environment
+from alf.trainers import policy_trainer
+from alf.utils import common, dist_utils
+import alf.utils.checkpoint_utils as ckpt_utils
+
+flags.DEFINE_string('root_dir', os.getenv('TEST_UNDECLARED_OUTPUTS_DIR'),
+                    'Root directory for writing logs/summaries/checkpoints.')
+flags.DEFINE_float('epsilon_greedy', 1., "probability of sampling action.")
+flags.DEFINE_integer('random_seed', None, "random seed")
```

Comment:
perhaps random_seed default shouldn't be None?  So default behavior will be consistent?

Reply:
My current thinking is that the test should always pass no matter what the seed is. If not, we can consider fixed seed in the future.###

Reply:
make sense.  given env randomness is gone, this should work.